Sources/SwiftMemcache/MemcachedValue.swift

```diff
@@ -33,14 +33,15 @@ extension MemcachedValue where Self: FixedWidthInteger {
     ///
     /// - Parameter buffer: The ByteBuffer to which the integer should be written.
     public func writeToBuffer(_ buffer: inout ByteBuffer) {
-        buffer.writeInteger(self)
+        buffer.writeString(String(self))
```

Comment:
This seems wrong. If you want to store an integer we shouldn't convert it to a string. Is this the problem you encountered where you set a key and then want to increment it but the type in memcache was wrong?

Reply:
we could also make use of `writeIntegerAsASCII` here.

Reply:
So the root problem here is that memcache is operating on the values interpreted as ASCII and not the raw byte values right? So decrement expects that the value for a key is an ASCII string representing an integer?

Reply:
Yes that is correct, since the protocol is text based its wanting to receive the values as ASCII strings. we can fall back on making use of `writeIntegerAsASCII` and then implement one of the possible solutions we outlined here #9 to improve the performance.